app/controllers/mixins/containers_common_mixin.rb

```diff
@@ -210,6 +211,14 @@ def process_check_compliance(model, ids)
     end
   end
 
+  def check_smart_roles
+    %w(SmartProxy SmartState).each do |role|
+      unless MiqServer.all.any? { |s| s.has_active_role?(role.downcase) }
```

Comment:
Checking the zone sounds like the right way to go, but I have not been involved much with SSA and will defer to @roliveri.